cmd/root.go

```diff
@@ -42,34 +41,29 @@ func CreateConfig() {
 	flag.Parse()
 
 	if cfgFile != "" {
-		log.Printf("reading configs from file: %v", cfgFile)
+		log.Printf("Reading configs from file: %v", cfgFile)
 		file, err := os.Open(cfgFile)
 		if err != nil {
-			log.Print("can't open config file: ", err)
+			log.Print("Can't open config file: ", err)
 		}
 		defer file.Close()
 		decoder := json.NewDecoder(file)
 		err = decoder.Decode(&opts)
 		if err != nil {
-			log.Print("can't decode config JSON: ", err)
+			log.Print("Can't decode config JSON: ", err)
 		}
 	}
-	flag.Parse()
```

Comment:
I remember this was called twice for a good reason but can't remember what that was...

It had to with the fact that configs are parsed both from cmd and the file and we need to make sure one takes precedence over the other. [This](https://github.com/cortexproject/cortex/pull/1997) issue is very similar and they solved the problem by using a different flagset.

Let me try to figure this out.